In [1]:
import requests #to request http/https "fetch the pages"
from bs4 import BeautifulSoup #to parse the requested html page

In [11]:
url = "https://www.yellowpages.com.eg/en/condensed-search/dentist?"
html = requests.get(url)
html

<Response [406]>

# as you see it returns 406 code which mean the site noticed I'm using requests so I will use an agent as a parameter to the get method
# https://deviceatlas.com/blog/list-of-user-agent-strings

In [12]:
url = "https://www.yellowpages.com.eg/en/condensed-search/dentist?"
html = requests.get(url, headers={"User-Agent": "Mozilla/5.0 (Linux; Android 6.0; HTC One M9 Build/MRA58K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.98 Mobile Safari/537.3"})
html

<Response [200]>

In [14]:
html.text

'<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-1479413-1"></script><script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag(\'js\', new Date()); gtag(\'config\', \'UA-1479413-1\');</script> <meta http-equiv="Content-Type" name="viewport" content="width=device-width, initial-scale=1"/> <meta name="theme-color" content="#ffcd00" /> <title>dentist Cairo, Alexandria, Hurghada, Luxor, Aswan, Egypt | Yellow.com.eg</title><meta name="keywords" content="dentist Cairo, Cairo dentist, dentist Alexandria, Alexandria dentist, dentist Hurghada, Hurghada dentist, dentist Luxor, Luxor dentist, dentist Aswan, Aswan dentist, Dentists, Dentist Clinic, Dental Clinics, Dentist Clinics, Dental Clinic, Advanced Dental Clinic, Dental Medicine Clinics, Dentalclinics dentist Cairo, Alexandria, Hurghada, Luxor, Aswan, Egypt | Yellow.com.eg"><meta name="description" content="Find dentist in Cairo, Alexandria, Hur

# now we requested the page 
# and here comes the part of parsing of the html to extract the data

In [15]:
soup = BeautifulSoup(html.text, 'html.parser')

In [20]:
dents = soup.find_all('div', attrs={'class': ['searchResultsDiv', 'content-widget']})

In [28]:
def get_name_address(elem):
    return [elem.find('a', attrs={'class': ['companyName', 'companyTitle']}).text, elem.find('a', attrs={'class': 'company_address'}).text]

In [29]:
get_name_address(dents[0])

['Dr. Mohamed Emad - Whity',
 ' 123 El Amira Diana Tower, Hassan El Maamoun St., Floor 6, Nasr city, Cairo. ']

In [30]:
import pandas as pd

In [31]:
ls = []
for dent in dents:
    ls.append(get_name_address(dent))
df = pd.DataFrame(ls, columns= ['Name', 'Address'])

In [32]:
df

Name  \
0                            Dr. Mohamed Emad - Whity   
1     Pearl Dental & Beauty Center Dr. Tamer El Gamal   
2                                            D - Care   
3                Capital Dental - Dr. Ahmed Al Kaffas   
4        Lebanon Dental Clinic - Dr. Soheir El Shafei   
5                     La Clinique Dr. Kareem El Faham   
6   The White Smile Dental Clinic - Dr. Ahmed Abdu...   
7                   Dr. Mohamed Mounir Dentist Clinic   
8                            Dr. Mohamed Emad - Whity   
9       Smile House Dental Center - Dr. Moataz Hamada   
10                          Muharam Bek Dental Center   
11                  Dr. Moataz Alkhawas Dental Center   
12                             El Safwa Dental Center   
13                           Smiley - Dr. Heba El Adl   
14                                    Sky Dent Clinic   
15                                Pearl Dental Clinic   
16                                 Hala Dental Center   
17                                Dental Icon Clinics   
18                                     El Ders Center   
19                  Dr. Alaa El Khamrey Dental Center   
20                        Dr. Amr Emera Dental Clinic   

                                              Address  
0    123 El Amira Diana Tower, Hassan El Maamoun S...  
1    80 El Khalifa El Maamoun ST. Roxi Square., He...  
2            120 El Merghany St., Heliopolis, Cairo.   
3           60 Abbas El Akkad St., Nasr city, Cairo.   
4        11 Lebanon St., Floor 4, Mohandeseen, Giza.   
5    7 Bahaa El Din El Ghatwary St., Floor 2, Semo...  
6    21 El Batal Ahmed Abdel Aziz St. Off Gamaet E...  
7    26 Safwet Regal El Aamal Bldg. 5, Block A, Fa...  
8    123 El Amira Diana Tower, Hassan El Maamoun S...  
9    6 Mohamed Galal St., El Korba, Heliopolis, Ca...  
10   85 Moharram Bey St., Kasr Karim Bldg., Floor ...  
11   15 El Nasr Rd. Rabaa El Adaweya Sq., Floor 5,...  
12   24 Ahmed Tayseer St. Intersection Of Gamal Ab...  
13   20 El Batal Ahmed Abdel Aziz St., El Hekma Me...  
14   40 Suez Canal St., El Kasr Tower, Moharram be...  
15   Abdel Salam Aref St., El Saleh Tower, Floor 2...  
16        Suez Canal St., El mansoura, El dakahleya.   
17   29 Darna Towers, Ring Rd., Zahraa El Maadi, M...  
18   44 Mansour St. Intersection Mostafa El Maragh...  
19   177 El Horaira Rd., Floor 3 Flat 15, El ibrah...  
20   80 El Zobat Bldg., , Mostafa kamel, Alexandria.